### PR TITLE
[sandbox] Basic booster to launch the given app with Firejail. JB#49488

### PR DESCRIPTION
**Pull request by abranson**

### CODE REVIEW

File: rpm/mapplauncherd-booster-firejail.spec [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Source0:    %{name}-%{version}.tar.bz2
+BuildRequires:  pkgconfig(Qt5Core)
+BuildRequires:  pkgconfig(Qt5Concurrent)
+BuildRequires:  mapplauncherd-devel >= 4.1.0
+BuildRequires:  pkgconfig(qdeclarative5-boostable)
+Requires:  mapplauncherd >= 4.1.0
+Requires:  systemd-user-session-targets
+Requires:  firejail
+Requires:  xdg-dbus-proxy
+
```
Comment (rainemak) [inline rpm/mapplauncherd-booster-firejail.spec:19]: %{!?qtc_make:%define qtc_make make}

After an empty line.

File: rpm/mapplauncherd-booster-firejail.spec [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+%description
+Proof-of-concept booster implementation for launching apps in a Firejail
+sandbox with optional privileges.
+
+%prep
+%setup -q -n %{name}-%{version}
```
Comment (rainemak) [inline rpm/mapplauncherd-booster-firejail.spec]: autosetup -p1 <= even though there are no patches

File: src/booster-firejail.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/***************************************************************************
+**
+** Copyright (C) 2014 - 2020 Jolla Ltd.
```
Comment (rainemak) [inline src/booster-firejail.h]: New derivative work. This line is not needed.

File: src/booster-firejail.cpp [hunk 4]
```diff
@@ -0,0 +1,207 @@
+/***************************************************************************
+**
+** Copyright (C) 2014 - 2020 Jolla Ltd.
```
Comment (rainemak) [inline src/booster-firejail.cpp]: Same here

File: rpm/mapplauncherd-booster-firejail.spec [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+%build
+%qmake5
+
+make %{?jobs:-j%jobs}
```
Comment (rainemak) [inline rpm/mapplauncherd-booster-firejail.spec]: %qtc_make %{?_smp_mflags}

File: rpm/mapplauncherd-booster-firejail.spec (hunk 2 above)
Comment (abranson) [reply rpm/mapplauncherd-booster-firejail.spec]: Yeah that's a nice default

File: booster-firejail.pro [hunk 6]
```diff
@@ -0,0 +1,28 @@
+TEMPLATE = app
+
+TARGET = booster-firejail
+QT += concurrent
+
+CONFIG += link_pkgconfig
+QT += feedback dbus sql waylandclient
+PKGCONFIG += mlite5 gio-2.0 glib-2.0 gmodule-2.0 gobject-2.0
```
Comment (rainemak) [inline booster-firejail.pro]: where is mlite5 used?

File: booster-firejail.pro [hunk 7]
```diff
@@ -0,0 +1,28 @@
+TEMPLATE = app
+
+TARGET = booster-firejail
+QT += concurrent
```
Comment (rainemak) [inline booster-firejail.pro]: Note seeing this either

Comment (rainemak) [reply booster-firejail.pro]: Ah, in the buildrequires... and this adds the linking

File: rpm/mapplauncherd-booster-firejail.spec (hunk 1 above)
Comment (rainemak) [reply rpm/mapplauncherd-booster-firejail.spec:19]: %{!?qtc_qmake5:%define qtc_qmake5 %qmake5}

File: rpm/mapplauncherd-booster-firejail.spec [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+%description
+Proof-of-concept booster implementation for launching apps in a Firejail
+sandbox with optional privileges.
+
+%prep
+%setup -q -n %{name}-%{version}
+
+%build
+%qmake5
```
Comment (rainemak) [inline rpm/mapplauncherd-booster-firejail.spec]: %qtc_qmake5

File: src/booster-firejail.h [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+    //! Disable assignment operator
+    FirejailBooster & operator= (const FirejailBooster & r);
+
+    static const string m_boosterType;
+
+    //! wait for socket connection
+    void accept();
+
+#ifdef UNIT_TEST
+    friend class Ut_FirejailBooster;
```
Comment (rainemak) [inline src/booster-firejail.h]: Something coming on top of this?

File: rpm/mapplauncherd-booster-firejail.spec (hunk 8 above)
Comment (abranson) [reply rpm/mapplauncherd-booster-firejail.spec]: with something like this?
%{!?qtc_qmake5:%define qtc_qmake5 qmake}

File: src/booster-firejail.h (hunk 9 above)
Comment (abranson) [reply src/booster-firejail.h]: Not unless it gets auto-generated by some testing framework. Remove?

File: booster-firejail.pro (hunk 7 above)
Comment (abranson) [reply booster-firejail.pro]: Removed

File: booster-firejail.pro (hunk 6 above)
Comment (abranson) [reply booster-firejail.pro]: Removed

File: rpm/mapplauncherd-booster-firejail.spec (hunk 5 above)
Comment (abranson) [reply rpm/mapplauncherd-booster-firejail.spec]: Done

File: src/booster-firejail.h (hunk 3 above)
Comment (abranson) [reply src/booster-firejail.h]: Removed from both

File: rpm/mapplauncherd-booster-firejail.spec (hunk 1 above)
Comment (abranson) [reply rpm/mapplauncherd-booster-firejail.spec:19]: Done

File: src/booster-firejail.cpp [hunk 10]
```diff
@@ -0,0 +1,206 @@
+/***************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC.
```
Comment (rainemak) [inline src/booster-firejail.cpp]: Nitpick... (c) => lower case

File: src/booster-firejail.h [hunk 11]
```diff
@@ -0,0 +1,52 @@
+/***************************************************************************
+**
+** Copyright (C) 2020 Open Mobile Platform LLC.
```
Comment (rainemak) [inline src/booster-firejail.h]: Same here.

**Review by rainemak — APPROVED** [review]
Approving already. Fix that small nitpick before merging.

**Pull request merged**